### ISSUE TITLE
"prefer-const" rule does not look for variable reassignment in child block level scope

### ISSUE THREAD

**Issue by njgraf512** [palantir/tslint#3299]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.4.2
- __Running TSLint via__: (pick one) CLI 
#### TypeScript code being linted

```ts
function sampleFunction(color: string): string {
  let sampleString: string = 'You picked a';
  if (color === 'green') {
    sampleString += ' green';
  } else if (color === 'red') {
    sampleString += ' red';
  } else if (color === 'blue') {
    sampleString += ' blue';
  }
  sampleString += ' button';
  return sampleString;
}

sampleFunction('blue');
```

with `tslint.json` configuration:

```json
{
    "defaultSeverity": "error",
    "extends": [
        "tslint:recommended"
    ],
    "jsRules": {},
    "rules": {
      "arrow-parens": false,
      "interface-name": false,
      "interface-over-type-literal": false,
      "max-classes-per-file": false,
      "member-access": false,
      "object-literal-shorthand": false,
      "only-arrow-functions": false,
      "quotemark": [true, "single", "jsx-double", "avoid-escape", "avoid-template"]
    },
    "rulesDirectory": []
}

```

#### Actual behavior
Identifier 'sampleString' is never reassigned; use 'const' instead of
    'let'.

             139 }
             140
             141 function sampleFunction(color: string): string {
    >>>      142   let sampleString: string = 'You picked a';
             143   if (color === 'green') {
             144     sampleString += ' green';
             145   } else if (color === 'red') {

#### Expected behavior
Expected that I can reassign the value of `sampleString` within the if blocks.  This same behavior can be observed if sampleString was simply declared at the top as `let sampleString;`, and the rule seems to fail regardless of the type of the reassigned variable (boolean, number, etc). In essence, the rule "prefer-const" does not seem to consider standard lexical scoping rules by looking into lower level scopes for possible variable reassignment. Any help or thoughts on this are definitely appreciated!

**Comment by ajafff** [comment]
Looking at child scopes to check if the variable is reassigned is the main business of the rule.
There's a test for that: https://github.com/palantir/tslint/blob/b80c5f2f68f54993aa1d060a886e5833b8077428/test/rules/prefer-const/default/test.ts.lint#L85-L96

If it doesn't work, there must be something completely wrong ... I guess you used yarn 1.1.0 to install tslint? If yes, this is a duplicate of #3251

**Comment by ajafff** [comment]
@njgraf512 If you used yarn 1.1.0 or 1.2.0 to install your dependencies, can you please update to yarn 1.2.1 and install your dependencies from scratch. That could resolve your problems

**Comment by njgraf512** [comment]
@ajafff, really appreciate the heads up. Upgraded yarn, and it seems to have resolved the problem.

**Comment by ajafff** [comment]
Great. I'll close this issue. If you still have issues, just leave a comment or open a new one.

**Comment by njgraf512** [comment]
Sounds good. Will do.
